bpf/headers/arguments.h

```diff
@@ -0,0 +1,88 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _ARGUMENTS_H_
+#define _ARGUMENTS_H_
+
+#include "common.h"
+#include "bpf_tracing.h"
+#include "bpf_helpers.h"
+#include <stdbool.h>
+
+// Injected in init
+volatile const bool is_registers_abi /* TODO: set this up based on instrumented application's go version >= 1.17 */ = 1;
```

Comment:
TODO: Add Go version detection of instrumented application to set this from userspace side.

Reply:
Awesome, we can remove this option and the code that uses the older ABI.